openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
I like the idea of having this method take a user object, but I'm not sure `_get_user()` is the right name for it at that point, as you've already got the user.  Maybe `_authorize_user()`?  In which case, it probably doesn't need to return the user object.  It could either return a `bool` stating whether the user is allowed, or it could return nothing, and raise a `UserNotAuthorized` exception if the user is unauthorized.  I think I prefer the None/Exception version.